lib/pure/hashes.nim

```diff
@@ -236,13 +233,13 @@ proc hashIgnoreCase*(x: string): Hash =
   runnableExamples:
     doAssert hashIgnoreCase("ABRAcaDABRA") == hashIgnoreCase("abRACAdabra")
 
-  var h: Hash = 0
-  for i in 0..x.len-1:
-    var c = x[i]
-    if c in {'A'..'Z'}:
-      c = chr(ord(c) + (ord('a') - ord('A'))) # toLower()
-    h = h !& ord(c)
-  result = !$h
+  var
+    lowerString = newStringOfCap(len(x))
+    i = 0
+  while i <= high(x):
+    lowerString.addLowercaseChar(x[i])
```

Comment:
I think this crashes if `x = ""`

`while i < x.len:` wouldn't crash

A test case for empty strings would be nice.

Reply:
no, high for empty strings produces -1, not 0.